src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11787,13 +11775,10 @@ private bool ScanExplicitlyTypedLambda(Precedence precedence)
                     }
 
                     // eat the parameter name.
-                    if (this.IsTrueIdentifier())
-                    {
-                        this.EatToken();
-                    }
+                    var identifier = this.IsTrueIdentifier() ? this.EatToken() : CreateMissingIdentifierToken();
```

Comment:
What's the impact of this change?

Reply:
I don't expect an observable impact here. It just gives us a token to attach the `!!` to when scanning. Alternatively, we could make `ParseParameterNullCheck` accept a maybe-null identifier and just avoid calling the methods to attach the skipped tokens when the identifier is null.